setup_ctp7.sh

```diff
@@ -197,94 +198,109 @@ then
+        tar xzf xhal.tgz
+        rm -rf xhal.zip
+    fi
 
-    find . -type d -print0 | xargs -0 -n1 chmod a+rx
-    find . -type f -print0 | xargs -0 -n1 chmod a+r
-    find bin -type f -print0 | xargs -0 -n1 chmod a+rx
-    find lib -type f -print0 | xargs -0 -n1 chmod a+rx
+    if [ -n "${ctp7modtag}" ]
+    then
+        curl -L ${GEMDAQ_DOWNLOAD_URL}/ctp7_modules-${ctp7modtag}.tgz -o ctp7_modules.tgz
+        tar xzf ctp7_modules.tgz
+        rm -rf ctp7_modules.tgz
+    fi
+    set +x
 
-    echo "wget https://raw.githubusercontent.com/cms-gem-daq-project/ctp7_modules/develop/conf/conf.txt"
-    wget https://raw.githubusercontent.com/cms-gem-daq-project/ctp7_modules/develop/conf/conf.txt
-    echo "cp conf.txt vfat3/conf.txt"
-    cp conf.txt vfat3/conf.txt
-    echo "rm -rf conf.txt"
+    ## Obsolete?
+    set -x
+    mkdir -p vfat3
+    curl -L https://raw.githubusercontent.com/cms-gem-daq-project/ctp7_modules/release/legacy-1.1/conf/conf.txt \
+         -o vfat3/conf.txt
+    rsync -ach --progress --partial --links mnt root@${ctp7host}:/
+    rsync -ach --progress --partial --links fw oh_fw scripts xml python gemloader vfat3 \
+         root@${ctp7host}:${CARD_GEMDAQ_DIR}/
+    set +x
 
-    echo "rsync -ach --progress --partial --links bin fw lib oh_fw scripts xml python vfat3 root@${ctp7host}:/mnt/persistent/gemdaq/"
-    rsync -ach --progress --partial --links bin fw lib oh_fw scripts xml python gemloader vfat3 root@${ctp7host}:/mnt/persistent/gemdaq/
-   
     echo "Update LMDB address table on the CTP7, make a new .pickle file and resync xml folder"
-    echo "cp xml/* $XHAL_ROOT/etc/"
-    cp xml/* $XHAL_ROOT/etc/
+    set -x
+    echo cp -rfp xml/* ${GEM_ADDRESS_TABLE_ROOT}/
+    set +x
+
     echo "Upload rpc modules and restart rpcsvc"
-    scp -r $CTP7_MOD_ROOT/lib/*.so root@${ctp7host}:/mnt/persistent/rpcmodules 
     ssh root@${ctp7host} killall rpcsvc
-    if [ -n "${newuser}" ]
+    ssh -t root@${ctp7host} cat /etc/passwd|egrep gemuser >/dev/null
+    if [ "$?" = "0" ]
     then
-        ssh -t ${newuser}@${ctp7host} 'sh -lic "rpcsvc"'
+        ssh -t gemuser@${ctp7host} 'sh -lic "rpcsvc"'
     else
-        ssh -t texas@${ctp7host} 'sh -lic "rpcsvc"'
+        echo "CTP7 gemuser account does not exist on ${ctp7host}"
+        usage
     fi
-    echo "python python/reg_interface/reg_interface.py -n ${ctp7host} -e update_lmdb /mnt/persistent/gemdaq/xml/gem_amc_top.xml"
-    python $XHAL_ROOT/python/reg_interface/reg_interface.py -n ${ctp7host} -e update_lmdb /mnt/persistent/gemdaq/xml/gem_amc_top.xml
-    cp $XHAL_ROOT/etc/gem_amc_top.pickle xml/gem_amc_top.pickle
-    rsync -ach --progress --partial --links xml root@${ctp7host}:/mnt/persistent/gemdaq/
+
+    pushd xml
+    set -x
+    python ${XHAL_ROOT}/bin/gem_reg.py -n ${ctp7host} \
+           -e update_lmdb ${CARD_GEMDAQ_DIR}/xml/gem_amc_top.xml
+    cp -rfp ${GEM_ADDRESS_TABLE_ROOT}/gem_amc_top.pickle gem_amc_top_v${ctp7fw//./_}.pickle
```

Comment:
this won't work as is due to different set of libraries used on the host and on the AMC. A `pickle` file has to be specifically altered. See p27 [here](https://github.com/cms-gem-daq-project/sw_utils/blob/develop/v3ElectronicsUserGuide.md#to-update-oh-fw-on-ctp7)

Reply:
sure, but the altered file *is* the one that gets produced here, no?
so it's just an additional manual step after this?

Reply:
yes, exactly. Btw, the host address table has to be updated too **and** the previous pickle file has to be deleted.